src/hooks/useCharactersApi.ts

```diff
@@ -31,27 +32,28 @@ const useCharactersApi = (searchParams) => {
     [searchParams],
   );
 
-  const { data, error, setSize, isValidating } = useSWRInfinite(
+  const { data, error, setSize, isValidating } = useSWRInfinite<IData, Error>(
     getKey,
     fetcher,
   );
 
   useEffect(() => {
-    if (!data) return;
-
+    if (!data?.[0]) return;
     if (data[0].error) {
       setCharacters([]);
       setHasNextPage(false);
       return;
     }
 
-    const allCharacters = data.flatMap((data) => data.results);
-    const count = data[0]?.info.count;
-    const nextPage = data[data.length - 1]?.info.next;
+    const allCharacters = data
+      .flatMap((data: IData) => data.results)
+      .filter((character): character is ICharacter => character !== undefined); //Filtering ensures that the allCharacters array contains only elements of type Character.
```

Comment:
[req] additional filtering is not necessary if the callback returns empty array when results is `undefined`
```suggestion
    const allCharacters = data.flatMap(({ results }: IData) => results || []);
```